### PR TITLE
Use idempotent requests

### PR DESCRIPTION
**Pull request by swiercz**
https://stripe.com/docs/api/idempotent_requests

### CODE REVIEW

File: aa_stripe/models.py [hunk 1]
```diff
@@ -475,7 +464,10 @@ def refund(self, amount_to_refund=None):
         if (amount_to_refund + self.amount_refunded) > self.amount:
             raise StripeMethodNotAllowed("Refunds exceed charge")
 
-        stripe_refund = stripe.Refund.create(charge=self.stripe_charge_id, amount=amount_to_refund)
+        idempotency_key = "{}-{}-{}-{}".format(self.object_id, self.content_type_id,
+                                               self.amount_refunded, amount_to_refund)
```
Comment (Holek) [inline aa_stripe/models.py]: I will steal that in PayPal <3

File: aa_stripe/models.py [hunk 2]
```diff
@@ -401,6 +401,7 @@ def charge(self):
         self.customer = customer
         if customer:
             metadata = {"object_id": self.object_id, "content_type_id": self.content_type_id}
+            idempotency_key = "{}-{}".format(metadata["object_id"], metadata["content_type_id"])
```
Comment (remik) [inline aa_stripe/models.py]: This is wrong because dunning flow is generating more than 1 charge objects for the same order.

Comment (swiercz) [reply aa_stripe/models.py]: @remik Do we have any kind of dunning id generated?

Comment (remik) [reply aa_stripe/models.py]: you can try charge attempt number but it is Ro specific so here need to be universal.

**Pull request merged**